src/Tasks/Microsoft.NET.Build.Tasks/RunReadyToRunCompiler.cs

```diff
@@ -208,6 +279,20 @@ private string GenerateCrossgen2ResponseFile()
             }
 
             result.AppendLine("-O");
+            
+            if (string.IsNullOrEmpty(jitPath))
```

Comment:
As David mentioned, it would be cleaner to use Crossgen2 version check instead of introducing more occurrences of this dirty check.

Reply:
Hmm, while you pointed out how to access CG2 version in PrepareReadyToRunCompilation, I don't see any easy way to detect it in RunReadyToRunCompiler so we'd need to pass the version (or at least a boolean flag stating whether the version is 5 or not) as an extra argument and synthesize it somehow in the Microsoft.NET.Publish.targets script - that sounds like a bit of an overshoot to me.

Reply:
You can set it as a metadata in `ValidateCrossgen2Support`:
```C#
Crossgen2Tool.SetMetadata(MetadataKeys.Version, crossgen2PackVersion);
```
then use it in RunReadyToRunCompiler.cs and Microsoft.NET.Publish.targets instead of the `JitPath` check.

Reply:
Fixed in 6th commit.